django/thunderstore/api/cyberstorm/views/packages.py

```diff
@@ -64,92 +61,110 @@ class PackageListResponseSerializer(serializers.Serializer):
+        response.data["next"] = next_url
 
-    def get(self, request, *args, **kwargs):
-        qp = PackageListRequestSerializer(data=request.query_params)
-        qp.is_valid(raise_exception=True)
-        params = qp.validated_data
+        return response
 
-        # To improve cacheability.
-        for value in params.values():
-            if isinstance(value, list):
-                value.sort()
+    def get_serializer(self, package_page: Page, **kwargs):
+        """
+        Augment the objects with information required by serializer.
 
-        package_qs = self._get_package_queryset()
-        package_qs = filter_deprecated(params["deprecated"], package_qs)
-        package_qs = filter_nsfw(params["nsfw"], package_qs)
-        package_qs = filter_in_categories(params["included_categories"], package_qs)
-        package_qs = filter_not_in_categories(params["excluded_categories"], package_qs)
-        package_qs = filter_by_section(params.get("section"), package_qs)
-        package_qs = filter_by_query(params.get("q"), package_qs)
-        package_qs = self._annotate_queryset(package_qs)
-        package_qs = self._order_queryset(params["ordering"], package_qs)
-
-        package_page = self._paginate(params, package_qs)
-        (prev_url, next_url) = self._get_sibling_pages(params, package_page)
-        packages = self._get_packages_dicts(package_page)
-
-        serializer = PackageListResponseSerializer(
-            {
-                "count": package_page.paginator.count,
-                "previous": prev_url,
-                "next": next_url,
-                "results": packages,
-            },
-        )
+        This is needed since the serializer returns community-related
+        information that is not directly available on a Package object.
 
-        return Response(serializer.data, status=status.HTTP_200_OK)
+        This is an awkward place to do this but ListApiView didn't seem
+        to contain more suitable way to adjust the items.
 
-    def _get_package_queryset(self) -> QuerySet[Package]:
-        community_id = self.kwargs["community_id"]
-        community = get_object_or_404(Community, identifier=community_id)
-        queryset = get_community_package_queryset(community)
+        Using Page as type overrides the default implementation, but is
+        accurate as long as serializer_class is defined. If not, it may
+        be any iterable, e.g. QuerySet.
+        """
+        package_list = self._get_packages_dicts(package_page)
+        return super().get_serializer(package_list, **kwargs)
+
+    def get_queryset(self) -> QuerySet[Package]:
+        queryset = get_community_package_queryset(self._get_community())
+        return self._annotate_queryset(queryset)
+
+    def filter_queryset(self, queryset: QuerySet[Package]) -> QuerySet[Package]:
+        community = self._get_community()
+        require_approval = community.require_package_listing_approval
+        queryset = self.get_queryset()
```

Comment:
Why do we override the queryset we get in as a paremeter?

Probably not a blocker in that the view is likely to function correctly as it's used now, but this doesn't seem correct

Reply:
That's probably unintentional, at least I don't recall there to be any need for this.

Reply:
I knew DRF wasn't perfect but this PR alone has given me two new reasons to think it's worse than before 😄